pkg/broker/ingress/handler.go

```diff
@@ -67,34 +66,16 @@ type handler struct {
 }
 
 // NewHandler creates a new ingress handler.
-func NewHandler(ctx context.Context, options ...HandlerOption) (*handler, error) {
-	h := &handler{
-		logger: logging.FromContext(ctx),
+func NewHandler(ctx context.Context, httpReceiver HttpMessageReceiver, decouple DecoupleSink) *Handler {
```

Comment:
With this simplification, I think we can use the `struct providers`: https://github.com/google/wire/blob/master/docs/guide.md#struct-providers

Not sure if it's recommended though.

Reply:
I think in this case I'd prefer to use a constructor so that the Handler fields can remain hidden.